### PR TITLE
*: fix extra warnings about MockTiFlash

### LINKED ISSUE #31909: TiDB reports "pollTiFlashReplicaStatus returns error" constantly
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Start TiDB server with the unistore storage.

### 2. What did you expect to see? (Required)

Less warnings.

### 3. What did you see instead (Required)

It prints every 2 seconds.

```console
[2022/01/23 23:27:53.012 +08:00] [WARN] [ddl_tiflash_api.go:428] ["pollTiFlashReplicaStatus returns error"] [error="MockTiFlash is not accessible"]
[2022/01/23 23:27:55.014 +08:00] [WARN] [ddl_tiflash_api.go:428] ["pollTiFlashReplicaStatus returns error"] [error="MockTiFlash is not accessible"]
[2022/01/23 23:27:57.014 +08:00] [WARN] [ddl_tiflash_api.go:428] ["pollTiFlashReplicaStatus returns error"] [error="MockTiFlash is not accessible"]
```

### 4. What is your TiDB version? (Required)

701f126b2

### PR DESCRIPTION
**Pull request by CalvinNeo**
Signed-off-by: CalvinNeo <calvinneo1995@gmail.com>

### What problem does this PR solve?

Issue Number: close #31909

Problem Summary:

### What is changed and how it works?

Do not print warning messages when MockTiFlash is not available.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_tiflash_api.go [hunk 1]
```diff
@@ -425,7 +425,12 @@ func (d *ddl) PollTiFlashRoutine() {
 				if d.ownerManager.IsOwner() {
 					_, err := d.pollTiFlashReplicaStatus(sctx, pollTiflashContext)
 					if err != nil {
-						logutil.BgLogger().Warn("pollTiFlashReplicaStatus returns error", zap.Error(err))
+						switch err.(type) {
+						case *infosync.MockTiFlashError:
+							// If we have not set up MockTiFlash instance, for those tests without TiFlash, just suppress.
+						default:
+							logutil.BgLogger().Warn("pollTiFlashReplicaStatus returns error", zap.Error(err))
+						}
```
Comment (tangenta) [inline ddl/ddl_tiflash_api.go:433]: Please consider adding a std error in `errno/errcode.go` and using `(e *Error) Equal` to compare the error.

Comment (CalvinNeo) [reply ddl/ddl_tiflash_api.go:433]: This error only happens in test

**Review by tangenta — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**